swift/ql/lib/upgrades/ba4171b90d0665b40e9e203bac9e3d4a0b2d03ec/integer_literal_exprs.ql

```diff
@@ -0,0 +1,13 @@
+class Element extends @element {
+  string toString() { none() }
+}
+
+from Element i, string value
+where
+  integer_literal_exprs(i, value) and
+  not exists(Element interpolated |
+    interpolated_string_literal_expr_interpolation_count_exprs(interpolated, i)
+    or
+    interpolated_string_literal_expr_literal_capacity_exprs(interpolated, i)
+  )
+select i, value
```

Comment:
So if I'm understanding correctly, the `interpolated_string_literal_expr_interpolation_count_exprs` and `interpolated_string_literal_expr_literal_capacity_exprs` didn't actually have numerical (`int`) values, but linked to expressions that were (always?) an integer literal holding that value?

Reply:
exactly, they were always integer literals, as documented in the Swift compiler here:

https://github.com/apple/swift/blob/f5361ed68f32163d580dabf7bb596ea39c6b27a5/include/swift/AST/Expr.h#L938C1-L946